### PR TITLE
Fix/tao 7509/gap match improvements

### PR DESCRIPTION
**Pull request by oatymart**
https://oat-sa.atlassian.net/browse/TAO-7509

Delete button didn't work for text gaps in the Gap Match interaction.

All the code is actually there, only the button selector looks wrong.

So far I didn't find a reason to keep the selector `:not([data-html-editable] *)` for interaction delete buttons.

TODO:
- [x] add a unit test, but there doesn't seem to anything which properly covers interaction authoring at the moment.
- [x] version bump

### CODE REVIEW

File: views/js/qtiCreator/widgets/choices/helpers/formElement.js [hunk 1]
```diff
@@ -8,9 +8,21 @@ define(['jquery'], function($) {
...
+                }
+                $('.qti-item').trigger('toolbarchange', {
+                    callee: 'formElementHelper'
+                });
+            };
+
             $shuffleToggle.off('mousedown').on('mousedown', function(e) {
-                e.stopPropagation();
                 var $icon = $(this).children();
+                e.stopPropagation();
```
Comment (ferenckiss89) [inline views/js/qtiCreator/widgets/choices/helpers/formElement.js]: any specific reason to change the order? even related settings can be always on the very top

File: views/js/qtiCreator/widgets/interactions/gapMatchInteraction/states/Correct.js [hunk 2]
```diff
@@ -15,44 +15,57 @@ define([
         var response = interaction.getResponseDeclaration();
 
         var corrects  = _.values(response.getCorrect());
-       
-        commonRenderer.resetResponse(interaction); 
+
+        var bodyLength = interaction.bdy.bdy.length;
```
Comment (ferenckiss89) [inline views/js/qtiCreator/widgets/interactions/gapMatchInteraction/states/Correct.js]: there is always a bdy.bdy?

File: views/js/qtiCreator/widgets/interactions/gapMatchInteraction/states/Correct.js [hunk 3]
```diff
@@ -15,44 +15,57 @@ define([
...
         //add a specific instruction
-        instructionMgr.appendInstruction(interaction, __('Please fill the gap with the correct choices.'));
-       
+        if (bodyLength === 0) { // no method?
+            instruction = __('This interaction has no text defined in question mode.');
+        }
+        else if (numberOfGaps === 0) {
+            instruction = __('Please add one or more gaps to the text in question mode.');
+        }
+        else {
```
Comment (ferenckiss89) [inline views/js/qtiCreator/widgets/interactions/gapMatchInteraction/states/Correct.js]: we don't even need this else here, instruction can have this value by default

File: views/js/qtiCreator/widgets/choices/helpers/formElement.js [hunk 4]
```diff
@@ -8,9 +8,21 @@ define(['jquery'], function($) {
                 interaction = choice.getInteraction(),
```
Comment (ferenckiss89) [inline views/js/qtiCreator/widgets/choices/helpers/formElement.js]: header is missing from this file

File: views/js/qtiCreator/widgets/interactions/gapMatchInteraction/states/Correct.js [hunk 5]
```diff
@@ -15,44 +15,57 @@ define([
         var response = interaction.getResponseDeclaration();
```
Comment (ferenckiss89) [inline views/js/qtiCreator/widgets/interactions/gapMatchInteraction/states/Correct.js]: header is missing

File: views/js/qtiCreator/widgets/choices/helpers/formElement.js [hunk 6]
```diff
@@ -8,9 +8,21 @@ define(['jquery'], function($) {
...
                 $shuffleToggle = $container.find('[data-role="shuffle-pin"]');
 
+            var _toggleVisibility = function(show) {
```
Comment (ferenckiss89) [inline views/js/qtiCreator/widgets/choices/helpers/formElement.js]: afaik we should not use anymore underscore in function/method names (pls check with @krampstudio )

File: views/js/qtiCreator/widgets/choices/helpers/formElement.js (hunk 1 above)
Comment (oatymart) [reply views/js/qtiCreator/widgets/choices/helpers/formElement.js]: Just to silence the ESLint error in this one file.

File: views/js/qtiCreator/widgets/interactions/gapMatchInteraction/states/Correct.js (hunk 2 above)
Comment (oatymart) [reply views/js/qtiCreator/widgets/interactions/gapMatchInteraction/states/Correct.js]: Good catch, safety checks added.

File: views/js/qtiCreator/widgets/choices/helpers/formElement.js (hunk 4 above)
Comment (oatymart) [reply views/js/qtiCreator/widgets/choices/helpers/formElement.js]: Added.

File: views/js/qtiCreator/widgets/interactions/gapMatchInteraction/states/Correct.js (hunk 5 above)
Comment (oatymart) [reply views/js/qtiCreator/widgets/interactions/gapMatchInteraction/states/Correct.js]: Added.

File: views/js/qtiCreator/widgets/choices/helpers/formElement.js (hunk 6 above)
Comment (oatymart) [reply views/js/qtiCreator/widgets/choices/helpers/formElement.js]: Renamed.

File: views/js/qtiCreator/widgets/interactions/gapMatchInteraction/states/Correct.js (hunk 3 above)
Comment (oatymart) [reply views/js/qtiCreator/widgets/interactions/gapMatchInteraction/states/Correct.js]: Could be... I like to have all the assignments to this var at the same level of indentation - neither way is bad, the way I see it.

**Comment by oatymart** [comment]
The unit tests are created and are in a passing state, but the ones for deleting choices and gaps are still to be completed.

**Pull request merged**